embedding/embedlite/embedshared/EmbedLiteViewChild.cpp

```diff
@@ -979,13 +979,15 @@ mozilla::ipc::IPCResult EmbedLiteViewChild::RecvHandleDoubleTap(const LayoutDevi
   RefPtr<Document> document = presShell->GetDocument();
   NS_ENSURE_TRUE(document && !document->Fullscreen(), IPC_OK());
 
-  CSSRect zoomToRect = CalculateRectToZoomTo(document, cssPoint);
-
-  uint32_t presShellId;
-  ViewID viewId;
-  if (APZCCallbackHelper::GetOrCreateScrollIdentifiers(
-      document->GetDocumentElement(), &presShellId, &viewId)) {
-    ZoomToRect(presShellId, viewId, zoomToRect);
+  if (nsLayoutUtils::AllowZoomingForDocument(document)) {
+    CSSRect zoomToRect = CalculateRectToZoomTo(document, cssPoint);
+    uint32_t presShellId;
+    ViewID viewId;
+    if (APZCCallbackHelper::GetOrCreateScrollIdentifiers(
+        document->GetDocumentElement(), &presShellId, &viewId)) {
+      LOGW("TAP: ZoomeToRect");
+      ZoomToRect(presShellId, viewId, zoomToRect);
```

Comment:
Out of curiosity, that's the reason why GetApzcTreeManager()->ZoomToRect cannot cope with this? So, going down here AsyncPanZoomController::ZoomToRect

Reply:
I'm probably misreading the code, and/or misunderstanding your comment, but doesn't this code end up doing exactly that? So, the inverted call stack here goes:
1. EmbedLiteViewChild::ZoomToRect()
2. PEmbedLiteVeiwChild::SendZoomToRect()
3. EmbedLiteViewParent::RecvZoomToRect()
4. ApzcTreeManager::ZoomToRect()
5. AsyncPanZoomController::ZoomToRect()

Reply:
Correct

Reply:
I just pondering why apzc is not triggering the zoomtherect